msfar merge

mfsar opened the pull request:
1) base.h -> zdefiniowanie SYSTEM_TICK_RATE_MS wraz z obsluga dla POSIX'a (wartosc dla niego 1000)
1a) Uzycie SYSTEM_TICK_RATE_MS zamiast SYSTEM_CONFIG_TICKS_1S (posix-system.c)
2) Zmiany argumentu funkcji smphr_init (uns. int value -> bool taken)
2a) Odpowiednie zmiany w wywolaniach powyzszej funkcji ( !!!  0 -> true, 1->false !!!, dodanie stdbool.h)
3) Usuniecie nieodpowiedniego rzutowania w freertos-mutex.c
4) Dodanie dwoch funkcji dla freertos'owej kolejki (queue_push_to_front, queue_isr_push)
5) Przesuniecie funkcji związanych z delay'mi i tickami do *-system.c
5a) Dodanie funkcji system_get_tick_count
6) Poprawka bledu w argumentach dla xTaskCreate (freertos-task.c)
7) Dodanie inicjalizacji timerów w system_init() (posix-system.c)
8) Małe poprawki w stylistyce
9) Dodanie obsługi timerów dla POSIX'a
9a) Calosc opiera sie na jednym semaforze ktory czeka odpowiedni czas,
    Wywolanie funkcji start, stop, reset zwalnia semafor
    Timery sa dodawane do listy zgodnie z czasem "wybudzenia" (nie musimy przeszukiwac listy)

mfsar committed:
Polaczenie wersji M. Smoczynskiego, J. Tymejczyka i M. Sarnowskiego

1) base.h -> zdefiniowanie SYSTEM_TICK_RATE_MS wraz z obsluga dla POSIX'a (wartosc dla niego 1000)
1a) Uzycie SYSTEM_TICK_RATE_MS zamiast SYSTEM_CONFIG_TICKS_1S (posix-system.c)
2) Zmiany argumentu funkcji smphr_init (uns. int value -> bool taken)
2a) Odpowiednie zmiany w wywolaniach powyzszej funkcji ( !!!  0 -> true, 1->false !!!, dodanie stdbool.h)
3) Usuniecie nieodpowiedniego rzutowania w freertos-mutex.c
4) Dodanie dwoch funkcji dla freertos'owej kolejki (queue_push_to_front, queue_isr_push)
5) Przesuniecie funkcji związanych z delay'mi i tickami do *-system.c
5a) Dodanie funkcji system_get_tick_count
6) Poprawka bledu w argumentach dla xTaskCreate (freertos-task.c)
7) Dodanie inicjalizacji timerów w system_init() (posix-system.c)
8) Małe poprawki w stylistyce
9) Dodanie obsługi timerów dla POSIX'a
9a) Calosc opiera sie na jednym semaforze ktory czeka odpowiedni czas,
    Wywolanie funkcji start, stop, reset zwalnia semafor
    Timery sa dodawane do listy zgodnie z czasem "wybudzenia" (nie musimy przeszukiwac listy)

mpsm commented on src/freertos-mutex.c:
```diff
@@ -6,7 +6,6 @@
 #include <FreeRTOS/semphr.h>
 
 #include <stdbool.h>
-#include <stddef.h>
```
NULL sie defined in stddef, you can't remove this include

mpsm commented on src/posix-timer.c:
```diff
@@ -0,0 +1,262 @@
+    }
+    return (true);
+}
+
```
there is a lot of new lines here - do we really need them all? ;)

mpsm commented on src/posix-timer.c:
```diff
@@ -0,0 +1,262 @@
+    }
+}
+
+//-------------------------------------------------------------------------------
```
// is a c++ style comment - I prefer using /\* ... */ style
I don't see purpose of this separator though

mpsm commented on src/posix-system.c:
```diff
@@ -21,15 +22,19 @@ static void *posix_task_wrapper(void *arg);
+
     memset(tasks, 0, sizeof(tasks));
+
+    tmr_init(&timer_task);
```
I think adding this timer should be configurable

mpsm commented:
Valgrind is very unhappy with these changes
==6352== HEAP SUMMARY:
==6352==     in use at exit: 120 bytes in 4 blocks
==6352==   total heap usage: 34 allocs, 30 frees, 4,246 bytes allocated
==6352== 
==6352== 16 bytes in 1 blocks are still reachable in loss record 1 of 4
==6352==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==6352==    by 0x40166F: tmr_init (posix-timer.c:31)
==6352==    by 0x40119D: system_init (posix-system.c:29)
==6352==    by 0x400C63: main (queues.c:45)
==6352== 
==6352== 32 bytes in 1 blocks are still reachable in loss record 2 of 4
==6352==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==6352==    by 0x401088: smphr_init (posix-smphr.c:17)
==6352==    by 0x401C75: mutex_init (posix-mutex.c:10)
==6352==    by 0x401651: tmr_init (posix-timer.c:28)
==6352==    by 0x40119D: system_init (posix-system.c:29)
==6352==    by 0x400C63: main (queues.c:45)
==6352== 
==6352== 32 bytes in 1 blocks are still reachable in loss record 3 of 4
==6352==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==6352==    by 0x401088: smphr_init (posix-smphr.c:17)
==6352==    by 0x401665: tmr_init (posix-timer.c:29)
==6352==    by 0x40119D: system_init (posix-system.c:29)
==6352==    by 0x400C63: main (queues.c:45)
==6352== 
==6352== 40 bytes in 1 blocks are still reachable in loss record 4 of 4
==6352==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==6352==    by 0x401687: tmr_init (posix-timer.c:32)
==6352==    by 0x40119D: system_init (posix-system.c:29)
==6352==    by 0x400C63: main (queues.c:45)

mfsar committed:
1) Add push_to_front function
2) In other functions changed:
	posix_queue_t *q = *(posix_queue_t **)queue;
	to posix_queue_t *q = (posix_queue_t *)queue;

mfsar committed:
Create system_delay function

mfsar committed:
In posix sem function change first parameter:
	from *(sem_t **)s
	to (sem_t *)*s

mfsar committed:
In calls functions (in posix) from queues.h, semaphore.h and mutex.h
(without *init()) changes the first parameter:
	 from &parameter
	 to parameter

mfsar committed:
Add two functions tmr_destroy(tmr_t t) and tmr_destroy_list(). They a…
…re fixing problem with memory leaks found by Valgrind.

mfsar committed:
1) Change tmr_create -> tmr_init
1a) Change tmr_init -> tmr_task_init
2) Change "magic 1" to TMR_CREATE_DEFAULT_PERIOD

mfsar committed:
1) Fixes in code style:
  - change mutex, semaphore, delete_from_timer_list, add_to_timer_list prefix to tmr_
  - insert spaces in (posix_timer_t*)
  - some smaller things

mfsar committed:
Changes according to use timer directly (not via pointer):
1) change the definition of tmr_t
2) changes in functions declaration
3) changes in freertos-timer.c (t.handle -> t)
4) changes in posix-timer.c
5) changes in examples
Additionally defines SYSTEM_CONFIG_USE_TIMERS, to choose if timer task should be created

mfsar committed:
Fixes to avoid implicit casting

mfsar committed:
Delete declaration of tmr_get_period
